File: proxy/tests/script_engine/start_invoked.lua
pcall(require, "../../script_libs/lua/roxy.lua")
local count = 0
---@type Extension
local start_invoked = {
	start = function()
		count = 10
	end,
	request = function(flow)
		flow.request.body.text = tostring(count)
		count = count + 1
	end,
	response = function(flow)
		flow.response.body.text = tostring(count)
	end,
}
Extensions = { start_invoked }

File: proxy/tests/script_engine/start_invoked.py
from roxy import Flow, Extension


class StartInvoked(Extension):
    def start(self):
        self.count = 10

    def request(self, flow):
        flow.request.body.text = str(self.count)
        self.count += 1

    def response(self, flow):
        flow.response.body.text = str(self.count)
        self.count += 1


Extensions = [StartInvoked()]
